# 월별 물동량 장기 성장/계절성/환적 구조 파악

부산항만공사에서 제공하고 있는 물동량 예측 기초 데이터를 기반으로 연간 성장률, 월별 계절지수, 환적비중의 장기 변화를 계산했습니다.


## 계절지수

계절지수는 특정 월의 물동량이 평소 수준과 비교했을 때 상대적으로 많은 달인지, 적은 달인지를 숫자로 나타낸 값입니다. 부산항의 월별 물동량을 살펴보면 모든 달에 비슷한 양의 화물이 처리되는 것은 아닙니다. 연초나 연말, 명절 전후, 제조업의 생산 일정, 수출입 기업의 재고 조정, 글로벌 해운 일정 등 여러 요인에 따라 어떤 달에는 물동량이 반복적으로 증가하고, 어떤 달에는 반복적으로 감소할 수 있습니다. 이렇게 매년 비슷한 시기에 반복해서 나타나는 월별 변동의 특징을 계절성이라고 하며, 그 계절성이 어느 정도인지를 숫자로 표현한 것이 계절지수입니다.

예를 들어 부산항의 최근 여러 해 데이터를 살펴보았더니 8월의 물동량이 매년 다른 달보다 조금 많은 경향이 있다고 가정해 보겠습니다. 단순히 8월 물동량의 평균만 계산하면 8월이 실제로 특별히 많은 달인지 판단하기 어렵습니다. 부산항 전체 물동량 자체가 장기적으로 성장하고 있을 수도 있기 때문입니다. 10년 전의 월평균 물동량과 최근의 월평균 물동량 수준이 다르다면 단순한 물동량 숫자만 비교해서는 장기 성장 때문에 커진 것인지, 8월이라는 계절적 특성 때문에 커진 것인지 구분하기 어렵습니다. 계절지수는 이런 문제를 줄이기 위해 각 월의 물동량을 그 시기의 일반적인 수준과 비교하여 상대적인 크기로 바꾸어 봅니다.

계절지수를 1을 기준으로 표현한다고 생각하면 이해하기 쉽습니다. 계절지수가 1.00인 달은 일반적인 월평균 수준과 거의 비슷한 달이라는 뜻입니다. 계절지수가 1.10이라면 그 달의 물동량이 평소보다 약 10% 높은 경향이 있다는 뜻이고, 0.92라면 평소보다 약 8% 낮은 경향이 있다는 뜻입니다. 어떤 분석에서는 이를 100을 기준으로 표현하기도 합니다. 이 경우 110은 평균보다 약 10% 높은 달, 92는 평균보다 약 8% 낮은 달을 의미합니다. 표현 방식만 다를 뿐 의미는 같습니다.

예를 들어 최근 여러 해의 부산항 월별 계절지수가 1월 0.91, 2월 0.94, 3월 1.03, 4월 1.02, 5월 1.01, 6월 1.04, 7월 1.05, 8월 1.07처럼 나타났다고 가정해 보겠습니다. 이 결과를 보고 단순히 "8월 물동량이 가장 많았다"라고 해석하는 것이 아니라, 여러 해를 반복해서 관찰했을 때 8월에는 부산항의 평상시 월간 수준보다 물동량이 약 7% 높은 경향이 있었다라고 해석할 수 있습니다. 반대로 1월의 계절지수가 0.91이라면 여러 해 동안 1월은 일반적인 월보다 약 9% 낮은 물동량을 보이는 경향이 있었다는 뜻입니다.

여기서 중요한 점은 계절지수가 특정 연도 특정 월의 실적을 설명하는 숫자가 아니라는 것입니다. 예를 들어 2025년 8월에 물동량이 많았다고 해서 바로 "8월은 성수기다"라고 판단할 수는 없습니다. 그해에만 발생한 특수한 사건 때문일 수도 있기 때문입니다. 반면 2016년부터 2025년까지처럼 여러 개의 완전한 연도를 확인했을 때 8월의 상대적 물동량이 반복해서 높았다면, 그때부터 8월에 어느 정도 계절성이 존재한다고 판단할 근거가 생깁니다.

이 때문에 이 실습에서 최근 10개 완전 연도를 사용하는 것도 의미가 있습니다. 한두 해만 사용하면 코로나19, 글로벌 물류대란, 항만 파업, 경기침체처럼 특정 연도에 발생한 충격이 계절성처럼 보일 가능성이 있습니다. 여러 해를 함께 사용하면 특정 연도에만 발생했던 우연한 변동의 영향은 어느 정도 줄어들고, 매년 반복되는 월별 패턴을 상대적으로 더 잘 확인할 수 있습니다. 또한 불완전한 연도, 예를 들어 1월부터 8월까지만 데이터가 있는 연도를 그대로 포함하면 9~12월 데이터가 존재하지 않기 때문에 월별 비교가 왜곡될 수 있습니다. 그래서 먼저 연도별 월 수를 확인하고 12개월이 모두 존재하는 완전 연도만 선택하는 과정이 필요합니다.

계절지수를 구하는 목적은 결국 "부산항의 물동량에는 매년 반복되는 월별 패턴이 존재하는가?"라는 질문에 답하기 위한 것입니다. 연간 총량과 전년 대비 성장률은 부산항 물동량이 장기적으로 성장했는지 감소했는지를 보여주지만, 그 안에서 1월부터 12월까지 물동량이 어떻게 반복적으로 움직이는지는 알려주지 않습니다. 반대로 계절지수를 계산하면 "어떤 달이 상대적으로 물동량이 집중되는가", "어떤 달이 상대적으로 한산한가", "이러한 차이가 매년 어느 정도 반복되는가"를 확인할 수 있습니다.

따라서 장기 성장과 계절성은 서로 다른 개념으로 구분해서 보아야 합니다. 부산항의 연간 물동량이 10년에 걸쳐 지속적으로 증가했다면 그것은 장기적인 성장 추세입니다. 그런데 그 성장 과정에서도 매년 1~2월에는 상대적으로 물동량이 낮고 7~9월에는 상대적으로 높아지는 현상이 반복된다면 그것이 계절성입니다. 쉽게 말해 장기 성장은 "시간이 지나면서 전체 규모가 커지고 있는가"를 보는 것이고, 계절지수는 "한 해 안에서 어느 달이 평소보다 많거나 적은가"를 보는 것입니다.

## 데이터 메타데이터

- 원본 파일명: 부산항만공사_물동량 예측.csv
- 전체 행: 1376
- 출처: [공공데이터 포털](https://www.data.go.kr/data/15147930/fileData.do)
- 비고
  - 물동량단위: ton
  - 수출입구분
    - II: 수입
    - IT: 수입환적
    - OO: 수출
    - OT: 수출환적


In [68]:
import pandas as pd
import numpy as np

In [69]:
# 쥬피터 노트북 파일에서 출력되는 Pandas의 데이터의 형식(format)을 지정합니다.
# 특정 열의 데이터의 소수점 자리수가 너무 길어 정규표현(과학적표현)으로 되고 있기 때문에, 소수점 3번째 자리수까지만 표현할 수 있도록 설정했습니다.
pd.set_option("display.float_format", lambda value: format(value, '.3f'))

In [70]:
FILENAME = 'transported_quantity_data.csv'
DATA_FILE_PATH = f'../data/{FILENAME}'

In [71]:
# 데이터 읽어오기
origin = pd.read_csv(
    DATA_FILE_PATH,
    encoding='cp949'
)

# 데이터 확인

데이터의 모양과 기술통계를 확인합니다.

In [72]:
origin.head()

,연도,월,연월,수출입구분,수출입구분설명,물동량
0,2003,2,2003-02,OT,수출환적,161074.500
1,2003,11,2003-11,IT,수입환적,170733.500
2,2006,1,2006-01,OT,수출환적,203348.250
3,2018,1,2018-01,OO,수출,413797.750
4,2013,6,2013-06,OO,수출,389650.250


## 데이터 형태 확인

In [73]:
origin.shape

(1376, 6)

In [74]:
origin.columns.to_list()

['연도', '월', '연월', '수출입구분', '수출입구분설명', '물동량']

In [75]:
origin.dtypes

연도           int64
월            int64
연월             str
수출입구분          str
수출입구분설명        str
물동량        float64
dtype: object

In [76]:
origin.info()

<class 'pandas.DataFrame'>
RangeIndex: 1376 entries, 0 to 1375
Data columns (total 6 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   연도       1376 non-null   int64  
 1   월        1376 non-null   int64  
 2   연월       1376 non-null   str    
 3   수출입구분    1376 non-null   str    
 4   수출입구분설명  1376 non-null   str    
 5   물동량      1376 non-null   float64
dtypes: float64(1), int64(2), str(3)
memory usage: 64.6 KB


## 결측치 확인

In [77]:
origin.isna().sum()

연도         0
월          0
연월         0
수출입구분      0
수출입구분설명    0
물동량        0
dtype: int64

## 기술통계

기술통계의 내용만 간략히 확인합니다.

In [78]:
origin.describe()

,연도,월,물동량
count,1376.000,1376.000,1376.000
mean,2009.983,6.471,324784.229
std,8.862,3.464,119286.831
min,1992.000,1.000,90300.000
25%,2003.000,3.000,222387.250
50%,2010.000,6.000,322195.500
75%,2017.250,9.000,420758.438
max,2025.000,12.000,635437.000


### 범주형 데이터 확인

In [79]:
origin['수출입구분'].unique()

<StringArray>
['OT', 'IT', 'OO', 'II']
Length: 4, dtype: str

In [80]:
origin['수출입구분'].nunique()

4

In [81]:
origin['수출입구분'].value_counts().sort_index()

수출입구분
II    398
IT    290
OO    398
OT    290
Name: count, dtype: int64

- 연도, 월, 물동량은 숫자형이고 연월은 문자열이므로 시계열 정렬과 기간 계산 전에 날짜형으로 변환해야 합니다.
- 수출/수입은 398행씩, 수출환적/수입환적은 290행씩이어서 범주 빈도가 다릅니다.

In [82]:
df = origin.copy()

# 데이터 정제

1. 분석에 필요한 형태로 데이터를 준비합니다.

In [83]:
# 연월을 날짜형 데이터로 변경
df['연월'] = pd.to_datetime(
    df['연월'],
    format='%Y-%m',
    errors='coerce'
)

df.dtypes

연도                  int64
월                   int64
연월         datetime64[us]
수출입구분                 str
수출입구분설명               str
물동량               float64
dtype: object

- 연도별 데이터의 완전성 확인

여기서의 완전성은 다른 데이터와 공정하게 비교할 수 있을 정도로 필요한 관측값이 모두 갖추어져있는가를 의미합니다.

In [84]:
# 완전연도(연도데이터의 완전성)
completeness = df.groupby('연도').agg(
    월수=('월', 'nunique'),
    구분수=('수출입구분설명', 'nunique'),
    행수=('물동량', 'size')
)

completeness

,월수,구분수,행수
연도,,,
1992,12,2,24
1993,12,2,24
1994,12,2,24
1995,12,2,24
1996,12,2,24
1997,12,2,24
1998,12,2,24
1999,12,2,24
2000,12,2,24


- 완전연도가 되기 위한 조건: 당해연도에 12개월이 정확히 포함되어있고, 4개의 물동량 구분과, 예상한 48개의 값(매월 4개 수출구분코드에 해당하는 데이터)가 포함되어있는지

In [85]:
has_12_month = completeness['월수'] == 12
has_4_categories = completeness['구분수'] == 4
has_48_rows = completeness['행수'] == 48

In [86]:
complete_years = completeness[
    has_12_month
    & has_4_categories
    & has_48_rows
]

complete_years

,월수,구분수,행수
연도,,,
2001,12,4,48
2002,12,4,48
2003,12,4,48
2004,12,4,48
2005,12,4,48
2006,12,4,48
2007,12,4,48
2008,12,4,48
2009,12,4,48


2. 데이터 품질 확인

In [87]:
expected = df.groupby(['연도', '월'])['수출입구분설명'].nunique()

expected

연도    월 
1992  1     2
      2     2
      3     2
      4     2
      5     2
           ..
2024  10    4
      11    4
      12    4
2025  1     4
      2     4
Name: 수출입구분설명, Length: 398, dtype: int64

In [88]:
quality_info = pd.Series({
    '연월 변환 실패 갯수': df['연월'].isna().sum(),
    '완전 중복 행': df.duplicated().sum(),
    '데이터가 48행이아닌 행 수': expected.ne(4).sum(),
    '완전 연도 수': len(complete_years),
    '불완전 연도 수': completeness.index.difference(complete_years.index).size,
})

quality_info

연월 변환 실패 갯수          0
완전 중복 행              0
데이터가 48행이아닌 행 수    108
완전 연도 수             24
불완전 연도 수            10
dtype: int64

In [89]:
completeness.head()

,월수,구분수,행수
연도,,,
1992,12,2,24
1993,12,2,24
1994,12,2,24
1995,12,2,24
1996,12,2,24


In [90]:
completeness.tail()

,월수,구분수,행수
연도,,,
2021,12,4,48
2022,12,4,48
2023,12,4,48
2024,12,4,48
2025,2,4,8


2025년에는 2개월 분량의 데이터만 수집된 것을 확인할 수 있습니다.

그렇다면 2025년과 같이 일부 월만 있는 연도를 연감 증감률 계산에서 제외해야하는데, 어떻게 할 것인지 결정해야 합니다.

In [98]:
annual = df.groupby('연도').agg(
    총물동량=('물동량', 'sum')
)

annual

,총물동량
연도,
1992,2635374.000
1993,2941235.000
1994,3582517.000
1995,4090333.750
1996,4362035.000
1997,4811161.500
1998,5311311.500
1999,5660882.000
2000,6382317.750


In [103]:
annual['완전연도'] = annual.index.isin(complete_years.index)

annual

,총물동량,완전연도
연도,,
1992,2635374.000,False
1993,2941235.000,False
1994,3582517.000,False
1995,4090333.750,False
1996,4362035.000,False
1997,4811161.500,False
1998,5311311.500,False
1999,5660882.000,False
2000,6382317.750,False


In [108]:
annual_complete = annual.loc[complete_years.index].copy()
annual_complete['전년대비증감률'] = annual_complete['총물동량'].pct_change()

annual_complete.tail(10)

,총물동량,완전연도,전년대비증감률
연도,,,
2015,19468724.500,True,0.042
2016,19456291.250,True,-0.001
2017,20493475.250,True,0.053
2018,21662576.000,True,0.057
2019,21992000.250,True,0.015
2020,21823995.000,True,-0.008
2021,22706129.750,True,0.040
2022,22078194.500,True,-0.028
2023,23153508.000,True,0.049


만일 2025년의 데이터가 여기 포함되어있었다면 2025년은 1월,2월에 해당하는 8개의 행만 있기 떄문에 연간 합계를 '24년 기준의 48행과 비교하면 큰 감소처럼 보였을 것입니다. 그렇게 되면 관측 기간의 차이로 인한 증감률이 마치 수요 충격처럼 보일 여지가 있기 때문에, 완전성이 있는 데이터만 포함하여 분석할 수 있도록 한 차례 정제한 것입니다.

# 데이터 분석

- 완전 연도의 총물동량과 전년 대비 증감률에서 최고/최저 성장연도는 언제인가?

In [112]:
growth = annual_complete.dropna(subset=['전년대비증감률'])
min_max = pd.DataFrame({
    '최고 성장': growth.loc[growth['전년대비증감률'].idxmax()],
    '최저 성장': growth.loc[growth['전년대비증감률'].idxmin()],
})

min_max

,최고 성장,최저 성장
총물동량,14194595.750,11980610.000
완전연도,True,True
전년대비증감률,0.185,-0.109


In [114]:
# min_max 전치
min_max = min_max.T

min_max

,총물동량,완전연도,전년대비증감률
최고 성장,14194595.750,True,0.185
최저 성장,11980610.000,True,-0.109


In [115]:
min_max['연도'] = [growth['전년대비증감률'].idxmax(), growth['전년대비증감률'].idxmin()]

min_max[['연도', '총물동량', '전년대비증감률']]

,연도,총물동량,전년대비증감률
최고 성장,2010,14194595.750,0.185
최저 성장,2009,11980610.000,-0.109


In [118]:
annual_complete.tail(12)

,총물동량,완전연도,전년대비증감률
연도,,,
2013,17686098.500,True,0.038
2014,18683283.250,True,0.056
2015,19468724.500,True,0.042
2016,19456291.250,True,-0.001
2017,20493475.250,True,0.053
2018,21662576.000,True,0.057
2019,21992000.250,True,0.015
2020,21823995.000,True,-0.008
2021,22706129.750,True,0.040


완전 연도 가운데 2001년 증가율은 약 26.5%로 가장 높고 2009년은 약 -10.9%로 가장 낮습니다. 데이터상 최근에는 2022년 약 -2.8% 이후 2023년 약 4.9%, 2024년 약 5.4% 증가했습니다.

극단 연도를 찾는 목적은 원인을 자동으로 확정하는 것이 아니라 외부 사건과 집계 변화 여부를 조사할 후보를 정하는 데 있습니다. 전년 대비 증감률은 기준 연도의 크기에 민감하므로 총물동량도 함께 제시했습니다. 한 해의 반등을 장기 성장 추세로 부르려면 이동기간 평균이나 여러 해의 연평균 증가율을 추가로 확인해야 합니다.

- 2015~2024년 계절지수와 10년단위(2001, 2010, 2020, 2024) 환적비중의 의미

In [122]:
# 최근 10년의 월별 물동량 추출
recent = df[df['연도'].between(2015, 2024)].groupby(['연도', '월'])['물동량'].sum().rename('월별합계').reset_index()

recent

,연도,월,월별합계
0,2015,1,1620144.750
1,2015,2,1535283.500
2,2015,3,1694094.250
3,2015,4,1669119.750
4,2015,5,1647131.000
...,...,...,...
115,2024,8,2052806.000
116,2024,9,1877152.000
117,2024,10,2049390.750
118,2024,11,1998100.750


In [125]:
# 각 연도별 월평균 계산
recent['해당연도월평균'] = recent.groupby('연도')['월별합계'].transform('mean')

# 계절지수 계산
# 계절지수 = 해당 월의 물동량 / 해당 연도의 월평균 물동량
# 계절지수는 물동량 크기가 아니라 당해년도의 평소 수준과 비교하여 특정 달에 상대적으로 높았는지 낮았는지를 나타냅니다.
recent['계절지수'] = recent['월별합계'] / recent['해당연도월평균']

recent

,연도,월,월별합계,해당연도월평균,계절지수
0,2015,1,1620144.750,1622393.708,0.999
1,2015,2,1535283.500,1622393.708,0.946
2,2015,3,1694094.250,1622393.708,1.044
3,2015,4,1669119.750,1622393.708,1.029
4,2015,5,1647131.000,1622393.708,1.015
...,...,...,...,...,...
115,2024,8,2052806.000,2033501.667,1.009
116,2024,9,1877152.000,2033501.667,0.923
117,2024,10,2049390.750,2033501.667,1.008
118,2024,11,1998100.750,2033501.667,0.983


In [127]:
# 모든 연도에 대한 월별 집수 재집계(aggregation)
seasonality = recent.groupby('월')['계절지수'].agg(['mean', 'median', 'std'])

seasonality

,mean,median,std
월,,,
1,0.985,0.982,0.035
2,0.925,0.924,0.028
3,1.039,1.037,0.035
4,1.019,1.022,0.023
5,1.032,1.039,0.032
6,0.999,0.998,0.019
7,1.025,1.035,0.036
8,0.995,1.000,0.021
9,0.960,0.965,0.052


- 평균
1을 기점으로 소수점 단위별로 커지고 작아진것을 의미합니다.
가령, 7월 평균이 1.025면, 각 연도의 평소 월 수준보다 약 2.5%높았다는 것을 의미하고, 9월같은 경우는 0.960이니, 평소 월 수준보다 약 4% 낮았다 라고 분석해볼 수 있습니다.

- 중앙값
중앙값은 10년간 계절지수를 크기순으로 정렬했을 때 가운데에 있는 수준을 보여줍니다.
현재 데이터에서는 1.039가 가장 높은 중앙값입니다. 평균은 이 값때문에 올라갈 수 있지만 중앙값은 상대적으로 영향을 덜 받습니다. 따라서 mean과 median을 같이 보면, 특정 몇 년 때문에 평균이 올라간것인지 아니면 대부분의 연도에서 실제로 높은 월인지 판단하는 근거가 됩니다.

- 표준편차
우리 데이터에서 표준편차는 당해 월의 계절성이 연도마다 얼마나 일정했는가를 판단하는 값으로 이해하면 됩니다. 표준편차가 작게나오면 해당 월은 물동량 패턴이 비교적 안정적으로 반복된다고 이해할 수 있고, 표준편차가 크면, 특정 달이 무조건 높은 달이다 라고 단정짓기 어려운것입니다.



특정 달이 높게 집계되었다고 해서, 휴가철 때문에 높다, 특정 수출품 때문에 높다 라고 단정짓는 것은 불가능합니다. 이러한 집계 내용은 단순히 언제 반복적으로 높고 낮은지를 발견하는 분석입니다.

이 분석을 기반으로 왜 이런 데이터가 나왔는지에 대한 합당한 자료는 추가적으로 조사하여 분석해야 하는것입니다.

In [128]:
trade = df.pivot_table(
    index='연도',
    columns='수출입구분설명',
    values='물동량',
    aggfunc='sum',
)

trade

수출입구분설명,수입,수입환적,수출,수출환적
연도,,,,
1992,1182679.000,NaN,1452695.000,NaN
1993,1400756.000,NaN,1540479.000,NaN
1994,1861341.000,NaN,1721176.000,NaN
1995,2177758.000,NaN,1912575.750,NaN
1996,2338534.000,NaN,2023501.000,NaN
1997,2614249.000,NaN,2196912.500,NaN
1998,2862064.750,NaN,2449246.750,NaN
1999,3189473.250,NaN,2471408.750,NaN
2000,3785717.000,NaN,2596600.750,NaN


In [130]:
trade['환적비중'] = trade.filter(like='환적').sum(axis=1) / trade.sum(axis=1)

trade

수출입구분설명,수입,수입환적,수출,수출환적,환적비중
연도,,,,,
1992,1182679.000,NaN,1452695.000,NaN,0.000
1993,1400756.000,NaN,1540479.000,NaN,0.000
1994,1861341.000,NaN,1721176.000,NaN,0.000
1995,2177758.000,NaN,1912575.750,NaN,0.000
1996,2338534.000,NaN,2023501.000,NaN,0.000
1997,2614249.000,NaN,2196912.500,NaN,0.000
1998,2862064.750,NaN,2449246.750,NaN,0.000
1999,3189473.250,NaN,2471408.750,NaN,0.000
2000,3785717.000,NaN,2596600.750,NaN,0.000


In [131]:
trade.loc[[2001, 2010, 2020, 2024]]

수출입구분설명,수입,수입환적,수출,수출환적,환적비중
연도,,,,,
2001,2571132.500,1530041.750,2558561.500,1413066.000,0.365
2010,3950133.250,3183850.000,3967771.750,3092840.750,0.442
2020,4852616.750,6034501.500,4951252.750,5985624.000,0.551
2024,5409691.250,6794732.250,5495144.750,6702451.750,0.553


최근 10년의 계절지수 평균은 3월 약 1.039로 가장 높고 2월 약 0.925로 가장 낮습니다. 즉 각 해의 평균적인 달과 비교할 때 3월은 약 3.9% 높고 2월은 약 7.5% 낮은 경향을 알 수 있습니다.

환적비중은 2001년 약 36.5%에서 2024년 약 55.3%로 높아져 부산항의 국제 환적 허브 역할이 강화되었다고 볼 수 있습니다.

# 결론

이번 파일에서는 연도별 월 수, 교역 구분 수, 행 수를 계산해 12개월과 네 구분이 모두 갖춰진 연도만 연간 성장 분석에 사용했습니다. 그 결과 2001년 약 26.5% 증가가 최고, 2009년 약 10.9% 감소가 최저이며 최근에는 2022년 감소 뒤 2023년과 2024년에 다시 증가한 것을 확인했습니다. 이 값들은 원인을 설명하는 답이 아니라 외부 사건과 집계 정의를 조사할 후보 연도가됩니다.

계절성은 최근 2015~2024년 각 해의 월합계를 그 해 월평균으로 나눈 뒤 같은 달끼리 평균하여 장기 성장 수준을 제거했습니다. 3월은 평균보다 약 3.9% 높고 2월은 약 7.5% 낮은것으로 확인했고, 영업일 수와 연휴 효과가 있을 수 있다고 예상해보았습니다.

수출입 구분을 합치면 환적비중은 2001년 약 36.5%에서 2024년 약 55.3%로 상승해 부산항의 허브 기능 강화를 알 수 있습니다. 이 파일만으로 미래를 예측할 수는 없으며 예측 모델에 넣기 전 완결성, 구조 변화, 계절성, 충격 연도를 먼저 이해해야 합니다.